read.pl

```diff
@@ -128,16 +128,18 @@ sub inner_read_report {
     if ( $_ =~ /^\s*(\d+) / ) {
       $report2taskid[$i] = $1;
       $taskid2report[$1] = $i;
+      audit("inner_read_report: report2taskid[$i]=$1, taskid2report[$1]=$i")
     } else {
       $report2taskid[$i] = $prev_id;
-      $taskid2report[$i] = $prev_id;
+      audit("inner_read_report: report2taskid[$i]=$prev_id")
     }
     $prev_id = $report2taskid[$i];
     $i++;
   }
   close(IN);
 
   if ( $#report_tokens > -1 ) {
+    # WHY?!!!!
```

Comment:
You're totally right.  Putting the comment like this is quite unhelpful, as I can't even rememeber now what is was about.  I had probably something to do with me trying to figure out what exactly the code is trying to accomplish here.